gobblin-compaction/src/main/java/gobblin/compaction/Compactor.java

```diff
@@ -26,5 +26,5 @@
    * A method for data compaction, e.g., merge a snapshot table and a delta table.
    * @throws IOException
    */
-  public void compact() throws IOException;
+  public void compact() throws Exception;
```

Comment:
Please don't use Exception in the throws clause as it defeats the purpose of static exception checking. List the type of exceptions that can happent.